File: src/task_886.rs
impl Solution {
    fn dfs(v: usize, color: i32, graph: &Vec<Vec<i32>>, colors: &mut Vec<i32>) -> bool {
        colors[v] = color;
        for &to in &graph[v] {
            if colors[to as usize] == -1 {
                if !Self::dfs(to as usize, 1 - color, graph, colors) {
                    return false
                }
            } else if colors[to as usize] == color {
                    return false
            }
        }
        return true
    }



    pub fn possible_bipartition(n: i32, dislikes: Vec<Vec<i32>>) -> bool {
        let mut graph: Vec<Vec<i32>> = vec![vec![]; n as usize];

        for i in 0..dislikes.len() {
            graph[dislikes[i][0] as usize - 1].push(dislikes[i][1] - 1);
            graph[dislikes[i][1] as usize - 1].push(dislikes[i][0] - 1);
        }

        let mut colors = vec![-1; n as usize];

        for i in 0..(n as usize) {
            if colors[i] == -1 {
                if !Self::dfs(i, 0, &graph, &mut colors) {
                    return false
                }
            }
        }

        return true
    }
}

pub struct Solution;

File: solutions/task_886.py
from collections import deque

class Solution:
    def possibleBipartition(self, n: int, dislikes: list[list[int]]) -> bool:
        graph = [[] for _ in range(n + 1)]
        for u, v in dislikes:
            graph[u].append(v)
            graph[v].append(u)
        group = [-1] * (n + 1)

        def bfs(start):
            queue = deque([start])
            group[start] = 0
            while queue:
                node = queue.popleft()
                for neighbour in graph[node]:
                    if group[neighbour] == -1:
                        group[neighbour] = 1 - group[node]
                        queue.append(neighbour)
                    elif group[neighbour] == group[node]:
                        return False
            return True

        for i in range(1, n + 1):
            if group[i] == -1:
                if not bfs(i):
                    return False
        return True